pkg/client/cache/fifo.go

```diff
@@ -39,6 +44,16 @@ type Queue interface {
 	HasSynced() bool
 }
 
+// Helper function for popping from Queue.
+func Pop(queue Queue) interface{} {
```

Comment:
I see, you're only using it for tests. Please update this comment or people are going to get bitten by this.